app/helpers/blacklight_helper.rb

```diff
@@ -144,25 +144,18 @@ def render_icon(var)
   end
 
   # solr fq field is field parameter provided unless id_nums value starts with BIB
+  # @param id_nums [Array<String>] ID's for the linked documents (neighbors)
+  # @param bib_id [String] ID for the document being accessed (root)
+  # @param field [String] field containing ID's to the neighboring documents
+  # @return [String] the markup for the linked records
   def linked_records(id_nums, bib_id, field)
-    fq = ''
-    id_nums.each do |n|
-      bib_match = /(?:^BIB)(.*)/.match(n)
-      solr_field = bib_match ? 'id' : field
-      n = bib_match[1] if bib_match
-      fq += "#{solr_field}:#{n} OR "
-    end
-    fq.chomp!(' OR ')
-    resp = get_fq_solr_response(fq)
-    req = JSON.parse(resp.body)
-    other_versions = []
-    req['response']['docs'].each do |record|
-      unless record['id'] == bib_id
-        title = record['title_display']
-        other_versions << link_to(title, solr_document_url(record['id']))
-      end
-    end
-    other_versions.empty? ? [] : [other_versions]
+    neighborhood = ::NeighborhoodService::DocumentNeighborhood.new(neighbors: id_nums,
```

Comment:
I'm not sure if `neighborhood` and `neighbors` are clear names here. Can we pick a name which is similar to where it's being used? Maybe `version_ids` and `VersionFetcher` or something?

Reply:
Or maybe even something relating to the fact that it's a `linked_record`

Reply:
This has been restructured as a `LinkedDocumentResolver`